### PR TITLE
minor fixes to make C semantics work more correctly

### PR DESCRIPTION
**Pull request by dwightguth**
@andreistefanescu @yilongli please review

### CODE REVIEW

**Comment by andreistefanescu** [comment]
lgtm

File: java-backend/src/main/java/org/kframework/backend/java/kil/KItemProjection.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public KItemProjection(Kind kind, Term term) {
 
     public Term evaluateProjection() {
         // TODO(AndreiS): hack until new KORE makes this class obsolete
-        if (!term.sort().equals(Sort.KITEM)) {
+        if (!term.sort().equals(Sort.KITEM) && !term.sort().equals(Sort.BAG) && !term.sort().equals(Sort.BAG_ITEM)) {
```
Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/kil/KItemProjection.java:26]: I don't understand this but I guess it doesn't matter

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicUnifier.java [hunk 2]
```diff
@@ -525,21 +526,26 @@ public void unify(BuiltinSet builtinSet, Term term) {
...
-            /*
-             * AndreiS: commented out the check below as matching might fail due
-             * to KItem < K < KList subsorting:
-             * !cell.contentKind().equals(otherCell.contentKind())
-             */
-            fail(cell, otherCell);
-        }
+            Cell<?> otherCell = (Cell<?>) term;
+            if (!cell.getLabel().equals(otherCell.getLabel())) {
+                /*
```
Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicUnifier.java:536]: this comment could be removed

**Comment by yilongli** [comment]
lgtm but Jenkins is not started for this pull request

**Comment by dwightguth** [comment]
Jenkins actually did run it just did not indicate as such on this page for some reason. Let me rerun it and see if it works this time.

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
@yilongli can you please review the new changes to this code?

**Comment by yilongli** [comment]
lgtm

**Pull request merged**